File: JAVA/is_armstrong.java
import java.util.Scanner;
import java.lang.Math;
class is_armstrong {
	public static void main (String [] args) {
		Scanner SC = new Scanner(System.in);
		int x;
		int results;
		System.out.println("User, enter a number: ");
		x = SC.nextInt();
		results = is_armstrong(x);
		if (results == 1) {
			System.out.println(x + " is an Armstrong");
		}
		else {
			System.out.println(x + " is NOT an Armstrong");
		}
	}
	public static int is_armstrong (int num) {
		int orig = num;
		int d = digit_counter(num);
		int sums = 0;
		while (num > 0) {
			int temp = num % 10;
			temp = (int)Math.pow(temp,d);
			sums += temp;
			num /= 10;
		}
		if (sums == orig) {
			return 1;
		}
		else {
			return 0;
		}
	}
	public static int digit_counter (int num) {
		int counter = 0;
		if (num == 0) {
			counter = 1;
		}
		else {
			while (num > 0) {
				counter++;
				num /= 10;
			}
		}
	return counter;
	}
}

File: C#/is_armstrong.cs
namespace aviv_yunker {
	class is_armstrong {
		public static void Main (string [] args) {
			string x;
			int num, num_dig, orig;
			int d = 0;
			int sums = 0;
			float temp;
			System.Console.WriteLine("User, enter a number: ");
			x = System.Console.ReadLine();
			num = int.Parse(x);

			num_dig = num;
			orig = num;

			if (num_dig == 0) {
				d = 1;
			}
			else {
				while (num_dig > 0) {
					d++;
					num_dig /= 10;
				}
			}
			while (num > 0) {
				temp = num % 10;
				temp = (float)(Math.Pow(temp,d));
				sums += temp;
				num /= 10;
			}
			if (sums == orig) {
				System.Console.WriteLine("{0} IS an Armstrong", orig);
			}
			else {
				System.Console.WriteLine("{0} is NOT an Armstrong", orig);
			}
		}
	}
}
